app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -247,6 +308,27 @@ window.GOVUK.analyticsGa4.analyticsModules = window.GOVUK.analyticsGa4.analytics
 
     getHostname: function () {
       return window.location.hostname
+    },
+
+    getProtocol: function () {
+      return window.location.protocol
+    },
+
+    removeCrossDomainParams: function (clickData) {
+      var url = clickData.url
+      if (url.indexOf('_ga') !== -1 || url.indexOf('_gl') !== -1) {
+        // _ga & _gl are values needed for cross domain tracking, but we don't want them included in our click tracking.
+        url = url.replace(/_ga=([^&]*)/, '')
+        url = url.replace(/_gl=([^&]*)/, '')
+
+        // The following code cleans up inconsistencies such as gov.uk/&&, gov.uk/?&hello=world, gov.uk/?, and gov.uk/&.
+        url = url.replaceAll(/(&&)+/g, '&')
+        url = url.replace('?&', '?')
```

Comment:
Just want to check... this line uses `replace` but the previous line uses `replaceAll` - is that okay? Will there only ever be on instance of `?&`?

Reply:
@andysellick I think the idea is there'd only be one instance of `?` in the URL, so `?&` would only happen once